Build fails on ubuntu 16.04

Build fails on ubuntu 16.04 with gcc5 with the following error:
```bash
[125/157] Compiling C++ object 'nntrainer/d48ed23@@nntrainer@sha/app_context.cpp.o'.
FAILED: nntrainer/d48ed23@@nntrainer@sha/app_context.cpp.o 
ccache c++ -Inntrainer/d48ed23@@nntrainer@sha -Inntrainer -I../nntrainer -Inntrainer/../api -I../nntrainer/../api -I../nntrainer/../api/ccapi/include -Inntrainer/dataset -I../nntrainer/dataset -Inntrainer/layers -I../nntrainer/layers -Inntrainer/models -I../nntrainer/models -Inntrainer/optimizers -I../nntrainer/optimizers -Inntrainer/tensor -I../nntrainer/tensor -Inntrainer/utils -I../nntrainer/utils -Inntrainer/graph -I../nntrainer/graph -I/usr/include/openblas -I/usr/include/nnstreamer -I/usr/include/opencv -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++14 -g -DMIN_CPP_VERSION=201402 -Wredundant-decls -Wwrite-strings -Wformat -Wformat-nonliteral -Wformat-security -Winit-self -Waddress -Wno-multichar -Wvla -Wpointer-arith -Wno-error=varargs -O2 -ftree-vectorize -DUSE_BLAS=1 -D__LOGGING__=1 -DENABLE_TEST=1 -DNNSTREAMER_AVAILABLE=1 -DENABLE_NNSTREAMER_BACKBONE=1 -DENABLE_TFLITE_BACKBONE=1 -DENABLE_DATA_AUGMENTATION_OPENCV=1 -fPIC -pthread -MD -MQ 'nntrainer/d48ed23@@nntrainer@sha/app_context.cpp.o' -MF 'nntrainer/d48ed23@@nntrainer@sha/app_context.cpp.o.d' -o 'nntrainer/d48ed23@@nntrainer@sha/app_context.cpp.o' -c /home/kparichay/FastDrive/Working/nnsuite/nntrainer_pk_3.git/nntrainer/app_context.cpp
/home/kparichay/FastDrive/Working/nnsuite/nntrainer_pk_3.git/nntrainer/app_context.cpp: In static member function ‘static nntrainer::AppContext& nntrainer::AppContext::Global()’:
/home/kparichay/FastDrive/Working/nnsuite/nntrainer_pk_3.git/nntrainer/app_context.cpp:110:37: error: ‘_1’ was not declared in this scope
       std::bind(add_default_object, _1, instance));
                                     ^
/home/kparichay/FastDrive/Working/nnsuite/nntrainer_pk_3.git/nntrainer/app_context.cpp:110:37: note: suggested alternative:
In file included from ../nntrainer/app_context.h:19:0,
                 from /home/kparichay/FastDrive/Working/nnsuite/nntrainer_pk_3.git/nntrainer/app_context.cpp:18:
/usr/include/c++/5/functional:782:34: note:   ‘std::placeholders::_1’
     extern const _Placeholder<1> _1;
                                  ^
[126/157] Compiling C++ object 'nntrainer/d48ed23@@nntrainer@sta/app_context.cpp.o'.
FAILED: nntrainer/d48ed23@@nntrainer@sta/app_context.cpp.o 
ccache c++ -Inntrainer/d48ed23@@nntrainer@sta -Inntrainer -I../nntrainer -Inntrainer/../api -I../nntrainer/../api -I../nntrainer/../api/ccapi/include -Inntrainer/dataset -I../nntrainer/dataset -Inntrainer/layers -I../nntrainer/layers -Inntrainer/models -I../nntrainer/models -Inntrainer/optimizers -I../nntrainer/optimizers -Inntrainer/tensor -I../nntrainer/tensor -Inntrainer/utils -I../nntrainer/utils -Inntrainer/graph -I../nntrainer/graph -I/usr/include/openblas -I/usr/include/nnstreamer -I/usr/include/opencv -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++14 -g -DMIN_CPP_VERSION=201402 -Wredundant-decls -Wwrite-strings -Wformat -Wformat-nonliteral -Wformat-security -Winit-self -Waddress -Wno-multichar -Wvla -Wpointer-arith -Wno-error=varargs -O2 -ftree-vectorize -DUSE_BLAS=1 -D__LOGGING__=1 -DENABLE_TEST=1 -DNNSTREAMER_AVAILABLE=1 -DENABLE_NNSTREAMER_BACKBONE=1 -DENABLE_TFLITE_BACKBONE=1 -DENABLE_DATA_AUGMENTATION_OPENCV=1 -fPIC -pthread -MD -MQ 'nntrainer/d48ed23@@nntrainer@sta/app_context.cpp.o' -MF 'nntrainer/d48ed23@@nntrainer@sta/app_context.cpp.o.d' -o 'nntrainer/d48ed23@@nntrainer@sta/app_context.cpp.o' -c /home/kparichay/FastDrive/Working/nnsuite/nntrainer_pk_3.git/nntrainer/app_context.cpp
/home/kparichay/FastDrive/Working/nnsuite/nntrainer_pk_3.git/nntrainer/app_context.cpp: In static member function ‘static nntrainer::AppContext& nntrainer::AppContext::Global()’:
/home/kparichay/FastDrive/Working/nnsuite/nntrainer_pk_3.git/nntrainer/app_context.cpp:110:37: error: ‘_1’ was not declared in this scope
       std::bind(add_default_object, _1, instance));
                                     ^
/home/kparichay/FastDrive/Working/nnsuite/nntrainer_pk_3.git/nntrainer/app_context.cpp:110:37: note: suggested alternative:
In file included from ../nntrainer/app_context.h:19:0,
                 from /home/kparichay/FastDrive/Working/nnsuite/nntrainer_pk_3.git/nntrainer/app_context.cpp:18:
/usr/include/c++/5/functional:782:34: note:   ‘std::placeholders::_1’
     extern const _Placeholder<1> _1;
                                  ^
ninja: build stopped: subcommand failed.

```

--- a/nntrainer/app_context.cpp
+++ b/nntrainer/app_context.cpp
@@ -106,7 +106,8 @@ static void add_default_object(AppContext &ac) {
 
 AppContext &AppContext::Global() {
   static AppContext instance;
-  std::call_once(global_app_context_init_flag, add_default_object, instance);
+  std::call_once(global_app_context_init_flag, add_default_object,
+                 std::ref(instance));
   return instance;
 }
 

